pike/model.py

```diff
@@ -1677,6 +1687,10 @@ def write_request(
             buffer=None,
             remaining_bytes=0,
             flags=0):
+        if buffer and not isinstance(buffer, bytes):
+            warnings.warn("buffer must be bytes, got {!r}, casting as str and encoding "
+                          "with 'ascii'".format(buffer), UnicodeWarning)
+            buffer = str(buffer).encode("ascii")
```

Comment:
The `str` cast will inject chars in py3:
``` python
>>> '1234'.encode('ascii')
b'1234'
>>> str(b'1234').encode('ascii')
b"b'1234'"
```

Reply:
i would think that the `isinstance` check would preclude this...

maybe i should explicitly check that buffer is `bytes` or `str`, and if it's not `bytes` then encode it with `ascii`. thoughts?

Reply:
> i would think that the `isinstance` check would preclude this...

Ah, looks like it should indeed 😅 
Looks like, for example, `buffer = []` would be left as is, though.

Reply:
fixed by making the type assertions stronger